File: Recipe.UI/src/app/models/recipe.ts
import { Ingredients } from './ingredients';
import { Instructions } from './instructions';
import { RecipeCategory } from './recipe-category';
import { RecipeCuisine } from './recipe-cuisine';
export class Recipe{
    id?: number;
    name: string = "";
    description: string  = "";
    notes?: string =  "";
    image?: string  = "";
    ingredients?: Ingredients[] = [];
    instructions?: Instructions[] = [];
    recipeCategory?: RecipeCategory[] = [];
    recipeCuisine?: RecipeCuisine[] = [];
}

File: Recipe.API.Library/Models/Recipe.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Recipe.API.Library.Models;

public class Recipe
{
    public int Id { get; set; }

    [Required]
    [MaxLength(150)]
    public string Name { get; set; }

    [Required]
    [MaxLength(500)]
    public string Description { get; set; }

    [MaxLength(500)]
    public string? Notes { get; set; }

    public string? Image { get; set; }

    public List<Ingredients> Ingredients { get; set; } = new List<Ingredients>();

    public List<Instructions> Instructions { get; set; } = new List<Instructions>();

    public IList<RecipeCategory> RecipeCategory { get; set; } 
}
